sdv/metadata/single_table.py

```diff
@@ -331,6 +331,14 @@ def _detect_columns(self, data):
                     f"Unsupported data type for column '{field}' (kind: {dtype})."
                     "The valid data types are: 'object', 'int', 'float', 'datetime', 'bool'."
                 )
+
+            # Set the first ID column we detect to be the primary key
+            if sdtype == 'id':
+                if self.primary_key is None:
+                    self.primary_key = field
+                else:
+                    sdtype = 'unknown'
```

Comment:
An alternative approach was to assign `unknown` in the `detect_sdtype_for...` methods and keep track of primary key candidates, but I felt those methods didn't need to be responsible for tracking the primary key candidates and I wanted to avoid passing a list along or having it as a instance variable